src/Xamarin.Android.Tools.AndroidSdk/JdkInfo.cs

```diff
@@ -125,29 +127,44 @@ void ValidateFile (string name, string path)
 				throw new ArgumentException ($"Could not find required file `{name}` within `{HomePath}`; is this a valid JDK?", "homePath");
 		}
 
-		static  Regex   VersionExtractor  = new Regex (@"(?<version>[\d]+(\.\d+)+)(_(?<patch>\d+))?", RegexOptions.Compiled);
+		static  Regex   NonDigitMatcher     = new Regex (@"[^\d]");
 
 		Version GetJavaVersion ()
 		{
 			string version = null;
-			if (!ReleaseProperties.TryGetValue ("JAVA_VERSION", out version)) {
-				if (GetJavaSettingsPropertyValue ("java.version", out string vs))
-					version = vs;
+			if (ReleaseProperties.TryGetValue ("JAVA_VERSION", out version) && !string.IsNullOrEmpty (version)) {
+				version = GetParsableVersion (version);
+				if (ReleaseProperties.TryGetValue ("BUILD_NUMBER", out var build) && !string.IsNullOrEmpty (build))
+					version += "." + build;
+			}
+			else if (GetJavaSettingsPropertyValue ("java.version", out version) && !string.IsNullOrEmpty (version)) {
+				version = GetParsableVersion (version);
+			}
```

Comment:
Could we do the `!string.IsNullOrEmpty` check when constructing `ReleaseProperties`?

Then wouldn't have to do this check so much here.

Reply:
Maybe?  However, consider:

https://github.com/xamarin/xamarin-android-tools/blob/master/src/Xamarin.Android.Tools.AndroidSdk/Tests/JdkInfoTests.cs#L47-L51

https://github.com/xamarin/xamarin-android-tools/blob/master/src/Xamarin.Android.Tools.AndroidSdk/Tests/JdkInfoTests.cs#L136-L144

If we did as you suggest, `JUST_A_KEY` wouldn't be present in `jdk.ReleaseProperties`.

Is that desirable?  I don't know.